edb/edgeql/parser/grammar/expressions.py

```diff
@@ -582,6 +585,35 @@ def reduce_SimpleShapePointer_REMASSIGN_Expr(self, *kids):
         )
 
 
+class OnConflictSpecifier(Nonterm):
+    def reduce_ON_Expr(self, *kids):
+        self.val = kids[1].val
+
+    def reduce_empty(self, *kids):
+        self.val = None
+
+
+class OnConflictElse(Nonterm):
+    def reduce_ELSE_Expr(self, *kids):
+        self.val = kids[1].val
+
+    def reduce_empty(self, *kids):
+        self.val = None
+
+
+class OnConflictCause(Nonterm):
+    def reduce_UNLESS_CONFLICT_OnConflictSpecifier_OnConflictElse(self, *kids):
```

Comment:
This allows `ELSE` without an explicit conflict specifier, I think we should not allow that (at least initially).